### ISSUE TITLE
Standardize user mention formatting

### ISSUE THREAD

**Issue by WeebHiroyuki** [Dragory/ZeppelinBot#289]
According to [LogsPlugin.ts#L123](https://github.com/Dragory/ZeppelinBot/blob/master/backend/src/plugins/Logs/LogsPlugin.ts#L123), `ping_user` is deprecated, implying that `allow_user_mentions` is the preferred option. Yet in [getLogMessage.ts#L92](https://github.com/Dragory/ZeppelinBot/blob/master/backend/src/plugins/Logs/util/getLogMessage.ts#L92), the user mention formatting result used in logs is determined by _both_ `ping_user` and `allow_user_mentions`.

If `ping_user` was internally marked as deprecated, why is it still being used to determine a user mention formatting result?
Instead, could we just standardize `allow_user_mentions` as the determining condition for whether or not `verboseUserMention()` is called instead of `verboseUserName()`?
```js
if (config.allow_user_mentions) {
  mentions.push(verboseUserMention(user));
} else {
  mentions.push(verboseUserName(user));
}
```

**Comment by xPolar** [comment]
I'm assuming `ping_user` is still in use even though it's deprecated for backwards compatibility and etc. I'd say there are still quite a few servers that use `ping_user` not knowing it's been deprecated, I'd say the Dragory and the other developers wouldn't want to make a breaking change such as that.

**Comment by WeebHiroyuki** [comment]
I wouldn't doubt that it's not still used. However, it can be a bit misleading and confusing if you haven't inspected the source code, that's all I'm saying. I'm seeking standardization. Furthermore, we shouldn't be afraid of removing deprecations. It's not really a "breaking change" either. The most you'll get are a few logs where the user is mentioned. Dragory can post an announcement as well, if required.

**Comment by Dragory** [comment]
Currently the main blocker for removing deprecations is the lack of a robust way to programmatically edit and migrate server configs (to remove/rename/etc. the deprecated options from them). There are some solutions I'm looking into; the main obstacle is preserving YAML formatting and comments. The [yawn-yaml](https://github.com/mohsen1/yawn-yaml) package looks very promising in that area.

**Comment by WeebHiroyuki** [comment]
Yeah, that's usually a problem with removing deprecations. Maybe you could officially announce it as deprecated along with plans to remove it with a couple months and suggest the alternative solution?
